test/SpokePool.Relay.ts

```diff
@@ -0,0 +1,200 @@
+import { expect } from "chai";
+import { Contract } from "ethers";
+import { ethers } from "hardhat";
+import { SignerWithAddress, seedWallet, toBN, toWei } from "./utils";
+import { deploySpokePoolTestHelperContracts, enableRoutes, getRelayHash } from "./SpokePool.Fixture";
+import { amountToSeedWallets, amountToDeposit, amountToRelay, amountToRelayNetFees } from "./constants";
+
+let spokePool: Contract, weth: Contract, erc20: Contract, destErc20: Contract;
+let depositor: SignerWithAddress, recipient: SignerWithAddress, relayer: SignerWithAddress;
+
+describe.only("SpokePool Relayer Logic", async function () {
+  beforeEach(async function () {
+    [depositor, recipient, relayer] = await ethers.getSigners();
+    ({ weth, erc20, spokePool, destErc20 } = await deploySpokePoolTestHelperContracts(depositor));
+
+    // mint some fresh tokens and deposit ETH for weth for depositor and relayer.
+    await seedWallet(depositor, [erc20], weth, amountToSeedWallets);
+    await seedWallet(relayer, [destErc20], weth, amountToSeedWallets);
+
+    // Approve spokepool to spend tokens
+    await erc20.connect(depositor).approve(spokePool.address, amountToDeposit);
+    await weth.connect(depositor).approve(spokePool.address, amountToDeposit);
+    await destErc20.connect(relayer).approve(spokePool.address, amountToRelay);
+    await weth.connect(relayer).approve(spokePool.address, amountToRelay);
+
+    // Whitelist origin token => destination chain ID routes:
+    await enableRoutes(spokePool, [
+      {
+        originToken: erc20.address,
+      },
+      {
+        originToken: weth.address,
+      },
+    ]);
+  });
+  it("Relaying ERC20 tokens correctly pulls tokens net fees and changes contract state", async function () {
+    const { relayHash, relayData } = getRelayHash(
+        depositor.address,
+        recipient.address,
+        0, // deposit ID
+        666, // origin chain ID
+        destErc20.address
+    )
+
+    await expect(
+      spokePool
+        .connect(relayer)
+        .fillRelay(
+            relayData[0],
```

Comment:
can you not just use the spread operator here? `...relayData`?

Reply:
I see in a few of these cases you are overriding the data within `relayData`. this could be solved by returning an object from `getRelayhash` then overriding the named props using `{...relayData, namedPropYouWantToOverride}` or something like this so we dont need to define these long lists of indexes.

wdyt, my goal is fewer lines here

Reply:
done